spring-context/src/main/java/org/springframework/format/datetime/standard/DurationFormatter.java

```diff
@@ -18,28 +18,79 @@
 
 import java.text.ParseException;
 import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 import java.util.Locale;
 
 import org.springframework.format.Formatter;
+import org.springframework.format.annotation.DurationFormat;
+import org.springframework.lang.Nullable;
 
 /**
  * {@link Formatter} implementation for a JSR-310 {@link Duration},
- * following JSR-310's parsing rules for a Duration.
+ * following JSR-310's parsing rules for a Duration by default and
+ * supporting additional {@code DurationFormat.Style} styles.
  *
  * @author Juergen Hoeller
  * @since 4.2.4
- * @see Duration#parse
+ * @see DurationFormatterUtils
+ * @see DurationFormat.Style
  */
-class DurationFormatter implements Formatter<Duration> {
+class DurationFormatter implements Formatter<Duration> { //TODO why is this one package-private ? make public and change since taglet ?
```

Comment:
@jhoeller I was wondering if there was a particular reason this `Formatter` has been kept package-private so far?